src/controllers/user/welcome.ts

```diff
@@ -20,6 +23,33 @@ export const getWelcomeController = async (
     });
     updateUserInAuthenticatedSession(req, user);
 
+    if (req.session.certificationDirigeantRequested) {
```

Comment:
issue: note that, if the user has already been welcomed for joining this organisation with classic methods, he should not see this page.

I'd rather put this logic in checkUserHasBeenGreetedForJoiningOrganizationMiddleware:

```
        if (!isEmpty(organizationThatNeedsGreetings)) {
          await greetForJoiningOrganization({
            user_id: getUserFromAuthenticatedSession(req).id,
            organization_id: organizationThatNeedsGreetings.id,
          });

          if (req.session.certificationDirigeantRequested) {
            return res.redirect(`/users/welcome-executive`);
          } else {
            return res.redirect(`/users/welcome`);
          }
        }
```

I found this more understandable, when the dirigeant should see the welcome page.

Reply:
- Use a /user/welcome/dirigeant page 
- Trigger from the checkUserHasBeenGreetedForJoiningOrganizationMiddleware
- remove the `req.session.certificationDirigeantRequested` test from the welcome controller 
- ensure to not display two welcome page